src/extendedstorageobserver.h

```diff
@@ -57,28 +58,32 @@ class MKCAL_EXPORT ExtendedStorageObserver //krazy:exclude=dpointer
        is being observed.
        @param info uids inserted/updated/deleted, modified file etc.
     */
-    virtual void storageModified(ExtendedStorage *storage, const QString &info) = 0;
+    virtual void storageModified(ExtendedStorage *storage, const QString &info);
 
     /**
-       Notify the Observer that a Storage is executing an action.
-       This callback is called typically for example every time
-       an incidence has been loaded.
+       Notify the Observer that a Storage has finished an action.
 
        @param storage is a pointer to the ExtendedStorage object that
        is being observed.
+       @param error true if action was unsuccessful; false otherwise
        @param info textual information
     */
-    virtual void storageProgress(ExtendedStorage *storage, const QString &info) = 0;
+    virtual void storageFinished(ExtendedStorage *storage, bool error, const QString &info);
 
     /**
-       Notify the Observer that a Storage has finished an action.
+       Notify the Observer that a Storage has been updated to reflect the
+       content of the associated calendar.
```

Comment:
So this would be specifically about the local changes saved to the database? In that case could slightly rephrase to avoid assumptions this would notify about external ones. And perhaps the other way around on storageModified.

Noted that now the local notebook adjustments probably don't signal anything. Not yet sure if that matters on the nemo plugin.

Reply:
I've rephrased both notification handler documentation to be explicit about when this or that signal is used.

> Noted that now the local notebook adjustments probably don't signal anything.

Actually they do, I've put a note in `SqliteStorage::modifyNotebook()` mentioning that the transactionId is increased, but not saved internally. So when the `fileChanged` callback is triggered, the modification is seen as an external modification and `storageModified` is emitted.